### PR TITLE
Admin UI: Show deletion process table and deletion process details for a identity

### PR DESCRIPTION
**Pull request by stamenione**
# Readiness checklist

-   [ ] I added/updated unit tests.
-   [ ] I added/updated integration tests.
-   [x] I ensured that the PR title is good enough for the changelog.
-   [x] I labeled the PR.

# Description

image

image

image

image

image

image

### CODE REVIEW

File: AdminUi/apps/admin_ui/lib/core/modals/confirmation_dialog.dart [hunk 1]
```diff
@@ -1,16 +1,16 @@
...
   required String title,
   required String message,
   required String actionText,
+  required String cancelActionText,
 }) async {
   final result = await showDialog<bool>(
     context: context,
-    builder: (BuildContext context) => _ConfirmationDialog(title: title, message: message, actionText: actionText),
+    builder: (BuildContext context) =>
+        _ConfirmationDialog(title: title, message: message, actionText: actionText, cancelActionText: cancelActionText),
```
Comment (jkoenig134) [inline AdminUi/apps/admin_ui/lib/core/modals/confirmation_dialog.dart:13]: missing trailing comma, please be more careful about this

File: AdminUi/apps/admin_ui/lib/core/modals/confirmation_dialog.dart [hunk 2]
```diff
@@ -33,7 +34,7 @@ class _ConfirmationDialog extends StatelessWidget {
       actions: [
         OutlinedButton(
           onPressed: () => Navigator.of(context).pop(false),
-          child: Text(context.l10n.cancel),
+          child: Text(cancelActionText),
```
Comment (jkoenig134) [inline AdminUi/apps/admin_ui/lib/core/modals/confirmation_dialog.dart:37]: I'd like to make the cancelActionText optional and default to context.l10n.cancel:

```suggestion
          child: Text(cancelActionText ?? context.l10n.cancel),
```

File: AdminUi/apps/admin_ui/lib/home/home.dart [hunk 3]
```diff
@@ -1,6 +1,7 @@
 export 'clients_overview/clients_overview.dart';
 export 'identities_overview/deletion_process_audit_log_details/deletion_process_audit_log_details.dart';
 export 'identities_overview/identities_overview.dart';
+export 'identity_details/deletion_process_table/deletion_process_details/deletion_process_details.dart';
```
Comment (jkoenig134) [inline AdminUi/apps/admin_ui/lib/home/home.dart:4]: why are the deletion_process_details located in the deletion_process_table? I'd move this next to the deletion_process_table instead

File: AdminUi/apps/admin_ui/lib/home/identity_details/deletion_process_table/deletion_process_details/deletion_process_details.dart [hunk 4]
```diff
@@ -0,0 +1,189 @@
...
+  Widget build(BuildContext context) {
+    if (_deletionProcessesDetails == null) return const Center(child: CircularProgressIndicator());
+
+    final deletionProcessDetails = _deletionProcessesDetails!;
+    return Padding(
+      padding: const EdgeInsets.only(bottom: 16),
+      child: Column(
+        crossAxisAlignment: CrossAxisAlignment.start,
+        children: [
+          if (Platform.isMacOS || Platform.isWindows)
```
Comment (jkoenig134) [inline AdminUi/apps/admin_ui/lib/home/identity_details/deletion_process_table/deletion_process_details/deletion_process_details.dart:42]: I've changed this in EVERY PR of you - **_don't use Platform!_**

File: AdminUi/apps/admin_ui/lib/home/identity_details/deletion_process_table/deletion_process_details/deletion_process_details.dart [hunk 5]
```diff
@@ -0,0 +1,189 @@
...
+
+  const DeletionProcessDetails({required this.address, required this.deletionProcessId, super.key});
+
+  @override
+  State<DeletionProcessDetails> createState() => _DeletionProcessDetailsState();
+}
+
+class _DeletionProcessDetailsState extends State<DeletionProcessDetails> {
+  IdentityDeletionProcessDetail? _deletionProcessesDetails;
+  List<IdentityDeletionProcessAuditLogEntry> auditLogs = [];
```
Comment (jkoenig134) [inline AdminUi/apps/admin_ui/lib/home/identity_details/deletion_process_table/deletion_process_details/deletion_process_details.dart:23]: why is this not private?

File: AdminUi/apps/admin_ui/lib/main.dart [hunk 6]
```diff
@@ -56,6 +56,14 @@ final _router = GoRouter(
...
               pageBuilder: (context, state) => NoTransitionPage(child: IdentityDetails(address: state.pathParameters['address']!)),
             ),
+            GoRoute(
+              parentNavigatorKey: _shellNavigatorKey,
+              path: ':address/deletion-process-details/:deletionProcessId',
+              pageBuilder: (context, state) => NoTransitionPage(
+                child:
+                    DeletionProcessDetails(address: state.pathParameters['address']!, deletionProcessId: state.pathParameters['deletionProcessId']!),
+              ),
+            ),
```
Comment (jkoenig134) [inline AdminUi/apps/admin_ui/lib/main.dart:66]: this should be a subroute of `:address` instead being next to `:address`

File: AdminUi/apps/admin_ui/lib/main.dart [hunk 7]
```diff
@@ -56,6 +56,14 @@ final _router = GoRouter(
               path: ':address',
               pageBuilder: (context, state) => NoTransitionPage(child: IdentityDetails(address: state.pathParameters['address']!)),
             ),
+            GoRoute(
+              parentNavigatorKey: _shellNavigatorKey,
+              path: ':address/deletion-process-details/:deletionProcessId',
+              pageBuilder: (context, state) => NoTransitionPage(
+                child:
+                    DeletionProcessDetails(address: state.pathParameters['address']!, deletionProcessId: state.pathParameters['deletionProcessId']!),
```
Comment (jkoenig134) [inline AdminUi/apps/admin_ui/lib/main.dart:64]: trailing comma....

File: AdminUi/packages/admin_api_sdk/lib/src/endpoints/identity_endpoint.dart [hunk 8]
```diff
@@ -53,6 +53,32 @@ class IdentitiesEndpoint extends Endpoint {
...
+  Future<ApiResponse<IdentityDeletionProcessDetail>> getIdentityDeletionProcessDetails({
+    required String address,
+    required String deletionProcessId,
+  }) =>
+      get(
+        '/api/v1/Identities/$address/DeletionProcesses/$deletionProcessId',
+        transformer: IdentityDeletionProcessDetail.fromJson,
+      );
+
+  Future<ApiResponse<void>> cancelDeletionProcessAsSupport({
```
Comment (jkoenig134) [inline AdminUi/packages/admin_api_sdk/lib/src/endpoints/identity_endpoint.dart:73]: as Support?

File: AdminUi/packages/admin_api_sdk/lib/src/endpoints/identity_endpoint.dart [hunk 9]
```diff
@@ -53,6 +53,32 @@ class IdentitiesEndpoint extends Endpoint {
...
 
+  Future<ApiResponse<List<IdentityDeletionProcess>>> getIdentityDeletionProcesses({
+    required String address,
+  }) =>
+      get(
+        '/api/v1/Identities/$address/DeletionProcesses',
+        transformer: (e) => (e as List).map(IdentityDeletionProcess.fromJson).toList(),
+      );
+
+  Future<ApiResponse<IdentityDeletionProcessDetail>> getIdentityDeletionProcessDetails({
```
Comment (jkoenig134) [inline AdminUi/packages/admin_api_sdk/lib/src/endpoints/identity_endpoint.dart:64]:
```suggestion
  Future<ApiResponse<IdentityDeletionProcessDetail>> getIdentityDeletionProcess({
```

File: AdminUi/packages/admin_api_types/lib/src/identities/identity_deletion_process.dart [hunk 10]
```diff
@@ -0,0 +1,58 @@
...
+    this.gracePeriodReminder2SentAt,
+    this.gracePeriodReminder3SentAt,
+  });
+
+  factory IdentityDeletionProcess.fromJson(dynamic json) => _$IdentityDeletionProcessFromJson(json as Map<String, dynamic>);
+  Map<String, dynamic> toJson() => _$IdentityDeletionProcessToJson(this);
+}
+
+@JsonSerializable()
+class IdentityDeletionProcessDetail {
```
Comment (jkoenig134) [inline AdminUi/packages/admin_api_types/lib/src/identities/identity_deletion_process.dart:43]: the detail has less details than the actual deletion process?

@tnotheis is this correct?

Comment (stamenione) [reply AdminUi/packages/admin_api_types/lib/src/identities/identity_deletion_process.dart:43]: It is not it has the same, but other details information are not used. I will add them to the type.

Comment (tnotheis) [reply AdminUi/packages/admin_api_types/lib/src/identities/identity_deletion_process.dart:43]: That's sad but true. ;) I already mentioned that last Monday in our meeting with Norbert.

Comment (jkoenig134) [reply AdminUi/packages/admin_api_types/lib/src/identities/identity_deletion_process.dart:43]: Why cant you change that @tnotheis?

Comment (tnotheis) [reply AdminUi/packages/admin_api_types/lib/src/identities/identity_deletion_process.dart:43]: We could. But that's gonna be part of a separate PR.

Comment (tnotheis) [reply AdminUi/packages/admin_api_types/lib/src/identities/identity_deletion_process.dart:43]: But the goal of the PR is to make the flutter Admin UI contain everything the angular one contained before. And the angular one did show exactly what we have here now.
So I would consider "showing more details" are separate feature.

Comment (jkoenig134) [reply AdminUi/packages/admin_api_types/lib/src/identities/identity_deletion_process.dart:43]: True, but I'd like both types to extend each other, we have to do this later then.

File: AdminUi/apps/admin_ui/lib/home/identity_details/deletion_process_table/deletion_process_details/deletion_process_details.dart (hunk 5 above)
Comment (stamenione) [reply AdminUi/apps/admin_ui/lib/home/identity_details/deletion_process_table/deletion_process_details/deletion_process_details.dart]: Forgot to add underscore.

**Pull request merged**